easybuild/easyblocks/g/gromacs.py

```diff
@@ -209,8 +211,64 @@ def configure_step(self):
                 # to avoid that GROMACS finds and uses a system-wide CUDA compiler
                 self.cfg.update('configopts', "-DGMX_GPU=OFF")
 
+        # check whether CK2K is loaded as a dependency
+        cp2k_root = get_software_root('CP2K')
+        if str(self.cfg['with_cp2k']).upper() == 'TRUE':
+            if not cp2k_root:
+                msg = "Compilation with CP2K was expicitly selected, but CP2K was not found."
+                raise EasyBuildError(msg)
+            self.log.info("Building with CP2K was requested.")
+        elif str(self.cfg['with_cp2k']).upper() == 'FALSE':
+            if cp2k_root:
+                self.log.info('CP2K was found, but compilation without CP2K has been requested.')
+            cp2k_root = None
+        elif str(self.cfg['with_cp2k']).upper() == 'AUTO' and cp2k_root:
```

Comment:
I understand the rationale for `'AUTO'` and testing against a string, but it would be less awkward with something like
```
if self.cfg['with_cp2k'] == 'auto' and cp2k_root: 
  ...
elif self.cfg['with_cp2k']:   # when it's true
  ... 
else:   # when it's false
  ... 
``` 

Converting booleans to string then taking upper case is.... well, weird.

Reply:
Actually... is there value for 'AUTO' ? That makes a recipe kindda unpredictable ?